sdk/azidentity/aad_identity_client.go

```diff
@@ -155,11 +227,12 @@ func (c *aadIdentityClient) createClientSecretAuthRequest(tenantID string, clien
 }
 
 func (c *aadIdentityClient) createClientCertificateAuthRequest(tenantID string, clientID string, clientCertificate string, scopes []string) (*azcore.Request, error) {
-	urlStr := c.options.AuthorityHost.String() + tenantID + tokenEndpoint
+	urlStr := c.options.AuthorityHost.String()
```

Comment:
Can this be removed now?

Reply:
Yes, I'll update the code wherever this happens

Reply:
Ok, so there is one problem with changing this to simply append information to the path of the request. Any subsequent calls to create a new request will also try to append information to the path of the AuthorityHost (that means we are changing the default value of the authority host on the client, I dont think that is the best idea). However, if we dont mind changing the path on the authority host then I could add a check to compare the current path in the client to an expected path.

Reply:
I'm leaning towards leaving it as it was, that is with a copy of the path of the authority host

Reply:
I think it would make the code a bit clunky to add some sort of check like: 
if !HasSuffix(c.options.AuthorityHost.Path, "oauth/etc../token") { 
    then append
}

I think creating a copy of the default authority host url is better, because we are also not changing the default value that we use for reference for all requests related to each credential.

Reply:
Will the value ever change?  If not can we set `AuthorityHost` to the correct value in the constructor?

Reply:
I think that can work for most credentials (although, in the future, if we decide to export tenantID or some other relevant field that affects the building of the URL then it wont work, but for now that is not a problem), but Device Code uses two different endpoints, so that one definitely cannot be created in the constructor.